c/httpserver.c

```diff
@@ -2759,97 +2753,69 @@ static int serviceAuthNativeWithSessionToken(HttpService *service, HttpRequest *
   int authDataFound = FALSE; 
   HttpHeader *authenticationHeader = getHeader(request,"Authorization");
   char *tokenCookieText = getCookieValue(request,SESSION_TOKEN_COOKIE_NAME);
-  if (traceAuth){
-    printf("serviceAuthNativeWithSessionToken: authenticationHeader 0x%p, extractFunction 0x%p\n",
-           authenticationHeader, service->authExtractionFunction);
-  }
+  AUTH_TRACE("serviceAuthNativeWithSessionToken: authenticationHeader 0x%p, "
+      "extractFunction 0x%p\n",
+       authenticationHeader,
+       service->authExtractionFunction);
 
   if (service->authExtractionFunction != NULL) {
     if (service->authExtractionFunction(service, request) == 0) {
       authDataFound = TRUE;
     }
   } else if (authenticationHeader) {
-    HttpHeader *authenticationHeader = getHeader(request,"Authorization"); 
-#ifdef DEBUG 
-    printf("safAuth: auth header = 0x%x\n",authenticationHeader); 
-    fflush(stdout); 
-#endif 
-    if (authenticationHeader){ 
-      if (extractBasicAuth(request,authenticationHeader)){ 
-#ifdef DEBUG 
-        printf("back inside safAuthenticate after call to extractBasicAuth\n"); 
-        fflush(stdout); 
-#endif 
-        authDataFound = TRUE; 
-      } 
+    DEBUG_TRACE("safAuth: auth header = 0x%x\n",authenticationHeader); 
+    if (extractBasicAuth(request, authenticationHeader)) {
+      DEBUG_TRACE("back inside safAuthenticate after call to extractBasicAuth\n");
+      authDataFound = TRUE;
     } 
   }
     
+  response->sessionCookie = NULL;
 
-  strupcase(request->username); /* upfold username */
-  char *username = request->username;
-
-  if (traceAuth){
-    printf("AUTH: tokenCookieText: %s\n",(tokenCookieText ? tokenCookieText : "<noAuthToken>"));
-  } 
+  AUTH_TRACE("AUTH: tokenCookieText: %s\n",(tokenCookieText ? tokenCookieText : "<noAuthToken>"));
   if (tokenCookieText){
     if (sessionTokenStillValid(service,request,tokenCookieText)){
-      if (traceAuth){
-        printf("auth cookie still good, renewing cookie\n");
-      }
-      char *sessionToken = generateSessionTokenKeyValue(service,request,request->username);
+      AUTH_TRACE("auth cookie still good, renewing cookie\n");
+      char *sessionToken = generateSessionTokenKeyValue(service, request,request->username);
       if (sessionToken == NULL) {
         return FALSE;
       }
-      addStringHeader(response,"Set-Cookie",sessionToken);
```

Comment:
This (and all other invocations in this function) will be ignored because setStatus() would clear all pending headers.